src/static_canvas.class.js

```diff
@@ -1399,7 +1399,14 @@
     createSVGFontFacesMarkup: function() {
       var markup = '', fontList = { }, obj, fontFamily,
           style, row, rowIndex, _char, charIndex, i, len,
-          fontPaths = fabric.fontPaths, objects = this._objects;
+          fontPaths = fabric.fontPaths, objects = [];
+      
+      this._objects.forEach(function add(object) {
+        objects.push(object);
+        if (object._objects) {
+          object._objects.forEach(add);
+        }
+      });
```

Comment:
```suggestion
          fontPaths = fabric.fontPaths, objects = this.objects;
      
      this._objects.forEach(function add(object) {
        if (object._objects) {
          objects = objects.concat(object._objects);
          object._objects.forEach(add);
        }
      });
```

i wonder if this version would be eventually more efficient, changing the array size once for all objects rather than each object. Just a curiosity.

Reply:
What do you mean changing the array size once and for all? Doesn't concatenating the objects array with all subObjects change the array size as well?